### PR TITLE
v6.7.0

### PR DESCRIPTION
**Pull request by flaviendelangle**

### CODE REVIEW

**Review by alexfauquette — APPROVED** [review]
👌

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,76 @@
...
+#### Changes
+
+- [DataGrid] Allow overflowing grid root element (#9179) @cherniavskii
+- [DataGrid] Fix module augmentation error when using `@mui/lab` (#9235) @cherniavskii
+- [DataGrid] Fix row with ids matching `Object` prototype (#9265) @romgrk
+- [DataGrid] Improve grouping performance for large datasets (#9200) @romgrk
+- [DataGrid] Increase threshold to trigger memory leak warning (#9263) @m4theushw
+- [DataGrid] Update data grid migration guide to include updated type (#9272) @MBilalShafi
+- [DataGridPremium] Remove last line break on clipboard paste (#9163) @cherniavskii
+- [DataGridPro] Improve header filter menu visuals (#9181) @MBilalShafi
```
Comment (MBilalShafi) [inline CHANGELOG.md:47]:
```suggestion
- [DataGridPro] Improve header filter menu visuals (#9181) @MBilalShafi
- [DataGridPremium] Remove last line break on clipboard paste (#9163) @cherniavskii
```

**Review by MBilalShafi — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,76 @@
...
+
+We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:
+
+- 🌍 Add Romanian (ro-RO) locale on the pickers
+- 🌍 Improve German (de-DE) locale on the pickers
+- 🌍 Improve Czech (cs-CZ), German (de-DE) and Turkish (tr-TR) locales on the data grid
+- 🚀 Performance improvements
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+- Improve the default `format` prop value on the pickers.
```
Comment (LukasTy) [inline CHANGELOG.md:18]: I believe that we usually put more important and noteworthy highlights before the general highlights.
What do you think about moving it as the first item and potentially adding some sort of an emoji (i.e. 🎁 ) for consistency? 🤔

**Review by LukasTy — APPROVED** [review]

**Review by romgrk — APPROVED** [review]
There is some CI linting finishing up for #9037, if it passes it can be merged & included.

**Comment by flaviendelangle** [comment]
@romgrk  no problem to wait for this PR :+1:

**Comment by romgrk** [comment]
Some minor issues with that PR still, we can proceed without it.

**Pull request merged**